### DISCUSSION TITLE
Proposal - <podcast:remoteItem> tag

### DISCUSSION THREAD

**Discussion by agates** [Podcastindex-org/podcast-namespace#481] [Tag Proposal]
# Proposal: \<podcast:remoteItem\> tag

The intent of this proposal is to provide a tag that allows another item to be referenced without outright copying it. This should be an item in another feed entirely.

## Cross-app playlists

This allows one to provide lists of other items which one does not own copyright to, such as playlists of various different types of audio/video mediums.  In combination with [list mediums](https://github.com/Podcastindex-org/podcast-namespace/issues/363), one can use a feed made entirely of `<podcast:remoteItem>` tags and share them in a manner that allows applications to recognize them as playlists.

```xml
<rss xmlns:podcast="https://github.com/Podcastindex-org/podcast-namespace/blob/main/docs/1.0.md" version="2.0">
    <channel>
        <title>My favorite v4v music</title>
        <description>This is a playlist of some of my favorite v4v music</description>
        <link>https://example.com/rss-music-playlist.xml</link>
        <docs>http://blogs.law.harvard.edu/tech/rss</docs>
        <language>en</language>
        <generator>Alecks Gates wrote this by hand in Joplin</generator>
        <pubDate>Thu, 27 Oct 2022 22:56:54 -0500</pubDate>
        <lastBuildDate>Thu, 27 Oct 2022 22:56:54 -0500</lastBuildDate>
        <podcast:medium>musicL</podcast:medium>
        <podcast:remoteItem itemGuid="https://podcastindex.org/podcast/4148683#1" feedGuid="a94f5cc9-8c58-55fc-91fe-a324087a655b" />
        <podcast:remoteItem itemGuid="https://podcastindex.org/podcast/4148683#3" feedGuid="a94f5cc9-8c58-55fc-91fe-a324087a655b" />
        <podcast:remoteItem itemGuid="tag:soundcloud,2010:tracks/319791095" feedGuid="a5ad6f3f-a279-504c-bc6a-30054e6b50e1" />
        <podcast:remoteItem itemGuid="tag:soundcloud,2010:tracks/319789777" feedGuid="a5ad6f3f-a279-504c-bc6a-30054e6b50e1" />
    </channel>
</rss>
```

## Cross-app clips

Using a the same concept as playlists, one can combine the `<podcast:remoteItem>` with `<podcast:soundBite>` to share user-generated clips as RSS feeds and similarly share them with in a manner other apps can recognize them as such.

```xml
<rss xmlns:podcast="https://github.com/Podcastindex-org/podcast-namespace/blob/main/docs/1.0.md" version="2.0">
    <channel>
        <title>agates' public clips on Podverse</title>
        <description>This is a playlist of podcast clips created by agates on Podverse</description>
        <link>https://podverse.fm/profile/ZTfsvQRLw</link>
        <docs>http://blogs.law.harvard.edu/tech/rss</docs>
        <language>en</language>
        <generator>Alecks Gates' handwritten python</generator>
        <pubDate>Fri, 20 May 2022 20:15:00 -0500</pubDate>
        <lastBuildDate>Fri, 20 May 2022 20:15:00 -0500</lastBuildDate>
        <podcast:medium>podcastL</podcast:medium>
        <podcast:remoteItem itemGuid="PC2040" feedGuid="917393e3-1b1e-5cef-ace4-edaa54e1f810" medium="podcast">
            <podcast:soundbite startTime="3707" duration="36">It was longer than a Glenn Greenwald article ... but I read this all the way through and enjoyed it</podcast:soundbite>
        </podcast:remoteItem>
        <podcast:remoteItem itemGuid="PC2077" feedGuid="917393e3-1b1e-5cef-ace4-edaa54e1f810" medium="podcast">
            <podcast:soundbite startTime="3834" duration="44">Adam Curry: The reason why we even can discuss this is because we have no business</podcast:soundbite>
        </podcast:remoteItem>
        <podcast:remoteItem itemGuid="PC2047" feedGuid="917393e3-1b1e-5cef-ace4-edaa54e1f810" medium="podcast">
            <podcast:soundbite startTime="1515" duration="177">Jack Dorsey: Business models enabled by bitcoin</podcast:soundbite>
        </podcast:remoteItem>
        <podcast:remoteItem itemGuid="Buzzsprout-9880236" feedGuid="396d9ae0-da7e-5557-b894-b606231fa3ea" medium="podcast">
            <podcast:soundbite startTime="1844" duration="33">"Linux" Unplugged shout out</podcast:soundbite>
        </podcast:remoteItem>
    </channel>
</rss>
```

## Definition

- **&lt;podcast:remoteItem itemGuid="(guid of the item referenced)" feedGuid="\[guid of the feed referenced\]" medium="\[medium type\]" /&gt;**
    Channel (optional | multiple)
    
    This element defines a remote item. One or more may be defined per channel.
    
    It refers to an item outside of the feed. This should only be used for "list" medium types . This **should not** utilize *any* value-related tags contained within the feed. Only the value tags on the referenced feed should be respected.
    
    - `itemGuid` (required) - The `<guid>` of the referenced item. If an item does not have a `<guid>`, it cannot be referenced.
    - `feedGuid` (required) - The [`<podcast:guid>`](https://github.com/Podcastindex-org/podcast-namespace/blob/main/docs/1.0.md#guid) of the referenced item.
    - `medium` (optional) - The medium of the referenced item. The list of allowable values is maintained in the [`<podcast:medium>`](https://github.com/Podcastindex-org/podcast-namespace/blob/main/docs/1.0.md#medium) specification. If not set, the default non-list medium type of the the feed the tag is defined in is assumed.  Cannot be `mixed`. This allows an application to determine its capability of how to handle the referenced item.

## Future stages

There's a lot of potential to use this to reference items for other reasons besides playlists and clips.  This could go under some kind of `<podcast:related>` tag, to reference items within the same feed or outside of the feed.  Combining it with with a to-be-defined `<podcast:channel>` is also a possibility.

### Citations

```xml
<podcast:guid>00000000-0000-0000-0000-000000000000</podcast:guid>
<item>
  <guid>SomeEpisode</guid>
</item>

<podcast:related>
    <podcast:remoteItem itemGuid="SomeEpisode" />
    <podcast:remoteItem itemGuid="PC20107" feedGuid="917393e3-1b1e-5cef-ace4-edaa54e1f810" medium="podcast" />
</podcast:related>
```

### References

It might also allow multiple items to be linked within the same feed in a way that does not otherwise fit within a structured season, such as a live stream of a podcast and an edited version of the same content.

```xml
<podcast:guid>00000000-0000-0000-0000-000000000000</podcast:guid>
<podcast:medium>video</podcast:medium>
<podcast:liveItem>
  <guid>SomeLiveEpisode</guid>
</podcast:liveItem>

<podcast:related>
    <podcast:remoteItem itemGuid="SomeLiveEpisode" type="live" feedGuid="00000000-0000-0000-0000-000000000000" medium="video" />
</podcast:related>
```

**Comment by ryan-lp** [comment]
> This allows one to provide lists of other items which one does not own copyright to, such as playlists of various different types of audio/video mediums.

Is a playlist a "derivative work" under copyright law?

Spotify's playlists might be legal only because their terms of service contain this clause:

> You retain ownership of your User Content when you post it to the Service. However, in order for us to make your User Content available on the Spotify Service, we do need a limited license from you to that User Content. Accordingly, you hereby grant to Spotify a non-exclusive, transferable, sub-licensable, royalty-free, fully paid, irrevocable, worldwide license to reproduce, make available, perform and display, translate, modify, **<ins>create derivative works from, distribute</ins>**, and otherwise use any such User Content through any medium, whether alone or in combination with other Content or materials, in any manner and by any means, method or technology, whether now known or hereafter created, in connection with the Spotify Service. Where applicable and to the extent permitted under applicable law, you also agree to waive, and not to enforce, any "moral rights" or equivalent rights, such as your right to be identified as the author of any User Content, including Feedback, and your right to object to derogatory treatment of such User Content.

Penalties for music copyright infringement are quite hefty.

Putting that issue aside, you might also want to specify the intended usage of this tag. For example, podcast apps should probably present the metadata of the remote feed, including the author, link, and any funding and value tags.

**Comment by jamescridland** [comment]
> Is a playlist a "derivative work" under copyright law?

Compilation CDs are copyright, or at least [the record companies think so](https://www.theverge.com/2013/9/4/4693828/spotify-sued-for-allowing-playlists-turns-out-you-can-copyright-a). The case didn't go to trial in the end, and [legal opinion is unsure about what would have happened](https://www.simkins.com/news/can-copyright-subsist-in-a-music-compilation-or-playlist). But a "database" is most certainly copyrightable, and an RSS feed is probably a database in the eyes of the law.

Back to `podcast:remoteItem` - perhaps a use-case for this is Podnews's [New Podcast Trailers](https://podnews.net/article/new-podcast-trailers), which is an RSS feed consisting of trailers for podcasts. I include all their episode notes, link directly to the audio itself, and show the artwork in the feed. Notwithstanding that, Apple didn't want to list the show, since it contained other peoples' copyrighted material (which is correct, it does). I wasn't expecting it to get into Apple Podcasts, to be honest, which is why I asked them to review it.

I'm a little unsure how `podcast:remoteItem` does anything different other than the RSS feed for _New Podcast Trailers_. In one sense it allows the creator more control - they can remove the episode item from their own RSS feed and it automatically disappears from here - but they can still remove the audio file anyway.

**Comment by thomasrynne** [comment]
If this tag is used to create playlists only podcast apps that support the new tag would be able to use the play list. In RSS terms the feed would be completely empty. I think so far the new podcast namespace tags have been enhancements to a working RSS feed but as described this tag would lead to an unusable feed for most people. Perhaps the remoteItem tag should go inside the normal item tag, with fallback behaviour defined within the item tag?

I do like the idea of directly linking to podcast episodes but I see that was raised in #304 so perhaps this is only for playlists?

**Comment by jamescridland** [comment]
[https://podnews.net/sitemap/trailers-ri.xml](https://podnews.net/sitemap/trailers-ri.xml) is a fully-working, daily-updated example feed for those wanting to test.

It's [a feed of new and notable podcast trailers](https://podnews.net/trailers) from Podnews. (That page contains a standard RSS feed with full enclosures).

My experience building it was relatively simple: but I would say that the standard RSS feed allows me to make a consistent list of shows, with a description that explains why individual shows have been chosen, including individual artwork items.

I don't believe that any copyright is being infringed in [my standard RSS feed](https://podnews.net/trailers) since the audio enclosure link points to the individual item. However, it might be interesting if people want to play with the remote:item tag to have a version that fully works (and that contains some weird-looking GUIDs!)

**Comment by daveajones** [comment]
> https://podnews.net/sitemap/trailers-ri.xml is a fully-working, daily-updated example feed for those wanting to test.

Thanks for doing this James.

**Comment by paulerskine** [comment]
I heard this discussed on [Podcasting 2.0 Ep 108](http://adam.curry.com/html/PC2010820221028Podca-48ckkP0TTQv4jNNbMTzNKfGpplBRgm.html) this week and curious if the following three items would need to be addressed:

1. **Value-related tags.** While I know the proposal indicates no value-related tags, they were discussed on the show. It was discussed a podcast could indicate "#%" goes to a remote link. Assuming the % would be split amongst all links, what happens when fractions of Sats are needed. Simple example: Podcast is set to 1% to remote links, podcast earns 100 Sats, and there are two links to that podcast, what happens? Can each receive 0.5 Sats? I know having two links to a podcast that earns 100 Sats is probably not realistic, but the something like this could happen at a larger scale. I bring this up because if there is a promise of "money" from one entity to another, and then that money doesn't show up or only a portion shows up, a legal issue that could arise. This may be a good reason to go with the definition and _not_ include any value related tags.

2. **Circular References / Infinite Loops.** What happens if you have two feeds that reference each other (A -> B and B -> A)? Is this just something the podcast players would need to deal with? More complicated is if they're referenced through a few podcasts (A -> B -> C -> D -> A) or even an odd internal reference (A -> B -> C -> D -> B).

3. **Block Tag validation.** Let's say a Podcast A has Hosting Service 1 blocked in their feed but Podcast B, which references A, is served by Hosting Service 1, what happens? This again gets more complicated in the A -> B -> C -> D link where any one of them could have a block tag that would need to be honored.

None of this is to say this is a bad idea, just items I'm wondering if they need to be thought about.

**Comment by jasonyork** [comment]
I've been interested in this proposal for a while because I think it would fit well with my podcast sharing app.  So, as a test, I built out an alternate feed generator to use `remoteItem`, similar to what @jamescridland did above.

I made a sample feed available here:
https://podshare.fm/feeds/7hxJB.remote.rss

The "standard" version of the same feed is here:
https://podshare.fm/feeds/7hxJB.rss

Implementing this on the producer side was pretty easy, but thinking about parsing this feed on the consumer / player side I had a few thoughts:

By only providing the `feedGuid`, it almost always requires the client to hit the podcastindex.org API to find the episode and/or keep a local db of podcasts.  A slight change to the proposal could provide the `feedUrl` as a required attribute, along with the `feedGuid` as a recommended attribute.  This gives the client the ability to fetch the feed directly to find the item if/when it is necessary and not necessarily require a dependency on the API.  For example:

```xml
<podcast:remoteItem itemGuid="PC20117" 
                    feedUrl="https://feeds.podcastindex.org/pc20.xml"
                    feedGuid="917393e3-1b1e-5cef-ace4-edaa54e1f810" />
```

Similar to @thomasrynne 's comment above, it does seem odd to have a separate feed just for a playlist.  Ideally, this would be an enhancement to a "regular" feed.   An end user likely won't know if their player supports this tag so for my use case it's difficult to provide two versions of the same feed and make sure the user picks the correct one.  Or maybe there would be a way to provide and "alternate source" tag in the primary feed that links to the "media list" version?

When a client does need to hit the podcastindex.org API to resolve the list, I could see a potential need to allow the client to supply a list of item and feed Urls/GUIDs for search so an entire list could be resolved with one request.  Otherwise, if you have a feed with 50 remoteItems it's 50 hits to the API.  Maybe that's premature optimization, but something to keep in mind.

Does anyone know if a player/client has attempted to implement this yet?  I think a lot would be learned about the feasibility of this proposal once it's attempted in a real client.  Thanks!

**Reply by thebells1111** [reply]
I agree the feedUrl seems ideal,  so the remoteItem can be fetched directly from the feed itself without reliance on the Podcast Index. I would also like to see something like enclosureUrl as a way to fetch the media file directly without needing to go to the feed. This allows for immediate playing of the file while the other resources are being fetched. So something like this, 

```
<podcast:remoteItem    
    feedUrl="https://feeds.podcastindex.org/pc20.xml"
    feedGuid="917393e3-1b1e-5cef-ace4-edaa54e1f810"
    itemGuid="PC20117" 
    itemUrl="http://www.wewillrockyou.com/rocking-song.mp3 
/>
```

The potential issue I see with this is broken links if the podcaster or musicians moves their feed or files to a different host.

**Comment by geeknews** [comment]
For those that are creating an alternate feed with a remoteid-only feed several questions

1. What do you do with that feed
2. How is the master feed going to reference the remote-id feed

**Reply by jasonyork** [reply]
@geeknews - I created the feed more as a proof-of-concept that could be used by someone updating an app to consume this type of feed.  As far as how a "master" feed would reference a remote-id feed - that's a good question.  I don't even know if that would be the desired intent.

**Reply by jamescridland** [reply]
> 1. What do you do with that feed

I coded it as a weekend relaxation project, and to give this group something to play with

> 2. How is the master feed going to reference the remote-id feed

It isn't designed to, as far as I'm aware. They're separate feeds (and/or items).

In this case, [my remote feed RSS feed](https://podnews.net/sitemap/trailers-ri.xml) does exactly the same job as [the full RSS feed](https://podnews.net/sitemap/trailers.xml) - both link directly to the audio from a third-party podcaster.

Advantages with my technique of linking directly to the audio are:
a) I can explain in the item title/description what I'm linking to and why I'm linking to it
b) I can lay out the description in a standard way to help the listener understand why I've curated these links
c) it works with 100% of available podcast apps

Advantages with the remoteItem feed:
a) um.... it shouldn't matter if the RSS feed changes?

I must be honest, I don't see the use-case for this feed format at all.

**Comment by thomasrynne** [comment]
What is the advantage of remoteItem over having hosts resolve remote items serverside and generate normal item item tags (as @jamescridland demonstrated above)?
Both versions require work by hosts but generating normal item tags means no work is required by app developers and all apps support the feed.

**Reply by thebells1111** [reply]
I think it's an ownership thing. It's one thing to create a link to someone else's file. It's another thing to take the items out of their file and paste them into your file. I'm not sure how any copyright laws would effect it.

**Reply by jamescridland** [reply]
In fact, [playlists are also subject to copyright](https://www.theverge.com/2013/9/4/4693828/spotify-sued-for-allowing-playlists-turns-out-you-can-copyright-a), just as any database is.

**Reply by thebells1111** [reply]
I wonder if a list of links that point to the different sources would be considered a playlist and subject to copyright, especially since a fundamental trait of the internet is linking to different files.

**Comment by thomasrynne** [comment]
I think the spec should give an indication of when apps are expected to re-resolve the referenced item.

Podcast apps already poll feeds but expectations regarding resources referenced by feeds are not clear. The primary motivation for polling a feed is to see new items in the feed. At the same the less common modifications to existing items are noticed. Polling of a feed containing remote items would capture any new remoteItems in the feed, but without an extra step any modifications to the referenced item will not be noticed.

In practise, the referenced item is unlikely to change, so there may be a reluctance to add the extra step, but without it a mistake (for example in the value tag) might never be fixed.

I don't know what the right balance is though. Perhaps the expectation should be that feeds referenced by remoteItems are polled as if they were a normal subscription

**Comment by daveajones** [comment]
Added to phase-6 branch:  https://github.com/Podcastindex-org/podcast-namespace/blob/phase-6/docs/1.0.md#remote-item
